### PR TITLE
 #164983459 Link categories to categories show

### PR DESCRIPTION
**Pull request by Luleherll**
### PT Story: Link Categories to Categories Show for the ADMIN
https://www.pivotaltracker.com/n/projects/1904891/stories/164983459

Changes proposed in this pull request:
1. Make categories link to the category details page for ADMIN users

Ready for review:
@patmbolger @thesuss @weedySeaDragon

### CODE REVIEW

File: config/database.yml [hunk 1]
```diff
@@ -9,10 +9,14 @@ default: &default
 development:
   <<: *default
   database: shf_project_development
+  username: shf
```
Comment (weedySeaDragon) [inline config/database.yml]: The changes in this file are ones that just _you_ need, so you shouldn't submit them.

(be sure **not** to add  this to the project `.gitignore` though)

File: features/admin_clicks_cat_to_see_details.feature [hunk 2]
```diff
@@ -0,0 +1,89 @@
+Feature: Link categories to categories show
```
Comment (weedySeaDragon) [inline features/admin_clicks_cat_to_see_details.feature]: add " for an Admin" to the feature description  (name) so it's clear right in that description that this is limited to the admin.

I know it states it in the "As an...." below, but I think it should be in the Feature name.
Others might disagree.

**Review by weedySeaDragon — CHANGES_REQUESTED** [review]
Need 1 change: see my comments in the `database.yml` file and 1 possible change (feature name/description)

File: features/admin_clicks_cat_to_see_details.feature [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    When I click on "Groomer"
+    Then I should see "Groomer"
+    And I should see "PsycGroomerCo"
+    And I should see "GroomerNext"
+    And I should see "kingGroomer"
+    And I should see "Groomer1"
+    And I should not see "stockholmCo"
+    And I should not see "Company7"
+    And I should not see "Company4"
+    And I should not see "Company5"
```
Comment (patmbolger) [inline features/admin_clicks_cat_to_see_details.feature:88]: We should also test the reverse cases - that is, a visitor (not logged in), and a member, _can not_ see the business category link when they view a company.

You can use the step at line 30 in `features/step_definitions/assertion_steps.rb` to check that:

`Then I should not see <link label> link`

File: config/database.yml [hunk 4]
```diff
@@ -9,10 +9,14 @@ default: &default
...
+  password: shfpassword
```
Comment (patmbolger) [inline config/database.yml]: I recommend _not_ using your specific `username` and `password` values for `development` and `test` DB's - as @weedySeaDragon noted, those cannot go into this file, and also doing so just complicates things (and for what purpose?).

**Review by patmbolger — CHANGES_REQUESTED** [review]
Please see comments.

File: config/database.yml (hunk 4 above)
Comment (Luleherll) [reply config/database.yml]: Thanks for pointing this out. I had trouble connecting to my DB and this was a workaround but I forgot to remove it. Let me work on it

**Comment by Luleherll** [comment]
@weedySeaDragon @patmbolger I have made some changes. Check it out

File: features/admin_clicks_cat_to_see_details.feature [hunk 5]
```diff
@@ -0,0 +1,101 @@
+Feature: Link categories to categories show for an Admin
```
Comment (weedySeaDragon) [inline features/admin_clicks_cat_to_see_details.feature:1]: Now that this feature covers more than just _admins_, I think we should adjust the feature description.

[This is a short description of what this whole file covers:  Think of it as how to describe what this file covers to someone that asks "what is the purpose of this file?"]

My suggestion: `Link categories to the categories show page for Admins only` 
(or `... only for Admins`)

Other suggestions or ideas for this?

**Review by weedySeaDragon — CHANGES_REQUESTED** [review]
See my comment about the Feature description.  Let's adjust that and then this looks good to me.
(Good catch [as usual :-) ] , @patmbolger, re: ensuring we have the scenarios for members & visitors)

You shouldn't have needed to do a `git force push` to get your latest changes in.  We can work together on this if it seems you can't get them in any other way. (Here or via Slack)

**Review by patmbolger — APPROVED** [review]
Looks good!

**Review by weedySeaDragon — APPROVED** [review]

**Pull request merged**